component/otelcol/processor/probabilistic_sampler/probabilistic_sampler_test.go

```diff
@@ -0,0 +1,251 @@
+//go:build !race
+
+package probabilistic_sampler
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log/level"
+	"github.com/grafana/agent/component/otelcol/internal/fakeconsumer"
+	"github.com/grafana/agent/pkg/util"
+
+	"github.com/grafana/agent/component/otelcol"
+	"github.com/grafana/agent/pkg/flow/componenttest"
+	"github.com/grafana/dskit/backoff"
+	"github.com/grafana/river"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+func TestBadRiverConfigNegativeSamplingRate(t *testing.T) {
```

Comment:
It might be easier to have table driven tests like the ones in the [spanmetrics connector](https://github.com/grafana/agent/blob/main/component/otelcol/connector/servicegraph/servicegraph_test.go).